libkineto/CMakeLists.txt

```diff
@@ -133,13 +133,91 @@ IF (NOT ROCM_SOURCE_DIR AND NOT ROCTRACER_INCLUDE_DIR)
     set(LIBKINETO_NOROCTRACER ON CACHE BOOL "" FORCE)
 endif()
 
+# Set LIBKINETO_NOXPUPTI to explicitly disable XPUPTI
+# Otherwise, XPUPTI is disabled if not found
+if(NOT LIBKINETO_NOXPUPTI)
+    if (NOT SYCL_INCLUDE_DIR OR NOT SYCL_LIBRARY)
+        # find sycl compiler
+        set(SYCL_ROOT "")
+        if(DEFINED ENV{SYCL_ROOT})
+            set(SYCL_ROOT $ENV{SYCL_ROOT})
+        elseif(DEFINED ENV{CMPLR_ROOT})
+            set(SYCL_ROOT $ENV{CMPLR_ROOT})
+        endif()
+
+        string(COMPARE EQUAL "${SYCL_ROOT}" "" nosyclfound)
+        if(nosyclfound)
+            message(FATAL_ERROR "Cannot find SYCL compiler!")
+        endif()
+
+        # Find include path from binary
+        find_file(
+          SYCL_INCLUDE_DIR
+          NAMES include
+          HINTS ${SYCL_ROOT}
+          NO_DEFAULT_PATH)
+
+        # Find include/sycl path from include path
+        find_file(
+          SYCL_INCLUDE_SYCL_DIR
+          NAMES sycl
+          HINTS ${SYCL_ROOT}/include/
+          NO_DEFAULT_PATH)
+
+        list(APPEND SYCL_INCLUDE_DIR ${SYCL_INCLUDE_SYCL_DIR})
+
+        # Find library directory from binary
+        find_file(
+          SYCL_LIBRARY_DIR
+          NAMES lib lib64
+          HINTS ${SYCL_ROOT}
+          NO_DEFAULT_PATH)
+
+        # Find SYCL library fullname
+        find_library(
+          SYCL_LIBRARY
+          NAMES sycl
+          HINTS ${SYCL_LIBRARY_DIR}
+          NO_DEFAULT_PATH)
+
+        if((NOT SYCL_INCLUDE_DIR) OR (NOT SYCL_LIBRARY_DIR) OR (NOT SYCL_LIBRARY))
+            message(FATAL_ERROR "SYCL library is incomplete!!")
+        endif()
+        message(INFO " SYCL_LIBRARY = ${SYCL_LIBRARY}")
+        message(INFO " SYCL_INCLUDE_DIR = ${SYCL_INCLUDE_DIR}")
+    else()
+        message(INFO " SYCL_LIBRARY = ${SYCL_LIBRARY}")
+        message(INFO " SYCL_INCLUDE_DIR = ${SYCL_INCLUDE_DIR}")
+    endif()
+
+    # find xpupti
+    find_package(Pti)
+    if(TARGET Pti::pti_view)
+        message(INFO "Found XPUPTI")
+        set(XPU_xpupti_LIBRARY ${SYCL_LIBRARY})
+        list(APPEND XPU_xpupti_LIBRARY Pti::pti_view)
+        message(INFO " XPU_xpupti_LIBRARY = ${XPU_xpupti_LIBRARY}")
+    else()
+        set(LIBKINETO_NOXPUPTI ON CACHE STRING "" FORCE)
+        set(XPU_xpupti_LIBRARY)
+        message(FATAL_ERROR "Could not find XPUPTI for building kineto")
+    endif()
+endif()
+
+IF (NOT SYCL_INCLUDE_DIR OR NOT XPU_xpupti_LIBRARY)
+    set(LIBKINETO_NOXPUPTI ON CACHE BOOL "" FORCE)
+endif()
```

Comment:
can we move all of above lines to src/plugin/xpupti?

Reply:
sounds like the better solution is, make xpupti as a separate static .a ?

Reply:
ok

Reply:
.a is not a MUST option. You can move these lines to separate cmakefile first.